src/cpp/client.cpp

```diff
@@ -1090,6 +1101,227 @@ void Client::save(std::string address)
+        throw SRParameterException("The src_name parameter cannot "\
+                                   "be an empty string.");
+    }
+
+    if (dest_name.size() == 0) {
+        throw SRParameterException("The dest_name parameter cannot "\
+                                   "be an empty string.");
+    }
+
+    if (src_name == dest_name) {
+        return;
+    }
+
+    copy_list(src_name, dest_name);
+    delete_list(src_name);
+}
+
+// Get the length of the list
+int Client::get_list_length(const std::string& list_name)
+{
+    // Build the list key
+    std::string list_key = _build_list_key(list_name, false);
+
+    // Build the command
+    SingleKeyCommand cmd;
+    cmd << "LLEN" << Keyfield(list_key);
+
+    // Run the command
+    CommandReply reply = _run(cmd);
+
+    // Check for errors and return value
+    if (reply.has_error() > 0)
+        throw SRRuntimeException("LLEN command failed. The list "\
+                                 "length could not be retrieved.");
+
+    if (reply.redis_reply_type() != "REDIS_REPLY_INTEGER")
+        throw SRRuntimeException("An unexpected type was returned for "
+                                 "for list length.");
+
+    int list_length = reply.integer();
+
+    if (list_length < 0)
+        throw SRRuntimeException("An invalid, negative value was "
+                                 "returned for list length.");
+
+    return list_length;
+}
+
+// Poll the list length
+bool Client::poll_list_length(const std::string& name, int list_length,
+                              int poll_frequency_ms, int num_tries)
+{
+    // Enforce positive list length
+    if (list_length < 0) {
+        throw SRParameterException("A positive value for list_length "\
+                                   "must be provided.");
+    }
+
+    // Check for the requested list length, return if found
+    for (int i = 0; i < num_tries; i++) {
+        if (get_list_length(name) >= list_length)
+            return true;
+        std::this_thread::sleep_for(std::chrono::milliseconds(poll_frequency_ms));
+    }
+
+    return false;
+}
+
+// Retrieve datasets in aggregation list
+std::vector<DataSet> Client::get_datasets_from_list(const std::string& list_name)
+{
+    if (list_name.size() == 0) {
+        throw SRParameterException("The list name must have length "\
+                                   "greater than zero");
+    }
+
+    return _get_dataset_list_range(list_name, 0, -1);
+}
+
+// Retrieve a subset of datsets in the aggregation list
+std::vector<DataSet> Client::get_dataset_list_range(const std::string& list_name,
+                                                    const int start_index,
+                                                    const int end_index)
+{
+    if (list_name.size() == 0) {
+        throw SRParameterException("The list name must have length "\
+                                   "greater than zero");
+    }
+
+    return _get_dataset_list_range(list_name, start_index, end_index);
+}
+
```

Comment:
A `pop_dataset_list` could be a useful feature to have if you have multiple consumers of an aggregated list.